service/history/workflow/mutable_state_impl.go

```diff
@@ -7548,6 +7548,27 @@ func (ms *MutableStateImpl) applyUpdatesToUpdateInfos(
 }
 
 func (ms *MutableStateImpl) syncExecutionInfo(current *persistencespb.WorkflowExecutionInfo, incoming *persistencespb.WorkflowExecutionInfo, isSnapshot bool) error {
+	if transitionhistory.Compare(current.WorkflowTaskLastUpdateVersionedTransition, incoming.WorkflowTaskLastUpdateVersionedTransition) < 0 {
```

Comment:
Need to use `!=` instead of `<` I think. That's what we do in other sync logic as well to handle the case where overall transition history is moving forward but VT for this particular state machine is becoming smaller.